tests/lib/tools/tests.invariant

```diff
@@ -122,6 +123,22 @@ check_leftover_defer_sh() {
 	fi
 }
 
+check_broken_snaps() {
+	n="$1" # invariant name
+	(
+		# awk is hard - this just gives us the names of snaps that are broken
+		for broke_snap in $(snap list --all | tail -n +2 | awk '{print $(NF) " " $1}' | grep 'broken ' | awk '{print $(NF)}' | sort | uniq); do
```

Comment:
This requires a running snapd instance, I wonder if it wouldn't be desirable to have a custom check that doesn't require snapd (this most likely means checking that snap.yaml is readable for all `/snap/*/` snaps?)?

Reply:
Well slightly later on during the restore we will remove all snaps by calling `snap remove` (that actually was what triggered me to want to write this since that snippet was failing because of broken snaps), so I think it's a fair requirement for snapd to be running after a test completes